libdnf-cli/progressbar/progress_bar.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "libdnf-cli/progressbar/progress_bar.hpp"
+
+
+namespace libdnf::cli::progressbar {
+
+
+ProgressBar::ProgressBar(int64_t total_ticks) : total_ticks{total_ticks} {}
+
+
+ProgressBar::ProgressBar(int64_t total_ticks, const std::string & description)
+    : total_ticks{total_ticks}
+    , description{description} {}
+
+
+void ProgressBar::set_ticks(int64_t value) {
+    if (is_finished()) {
+        return;
+    }
+    auto new_ticks = value;
+    if (total_ticks >= 0) {
+        new_ticks = std::min(new_ticks, total_ticks);
+    }
+    current_speed_window_ticks += new_ticks - ticks;
+    ticks = new_ticks;
+}
+
+
+void ProgressBar::add_ticks(int64_t value) {
+    set_ticks(ticks + value);
+}
+
+
+void ProgressBar::start() {
+    if (begin == std::chrono::system_clock::from_time_t(0)) {
+        begin = std::chrono::system_clock::now();
+        state = ProgressBarState::STARTED;
+    }
+}
+
+
+void ProgressBar::update() {
+    if (is_finished()) {
+        return;
+    }
+
+    if (ticks >= total_ticks) {
+        percent_done = 100;
+    } else if (total_ticks == 0) {
+        // can't divide by zero, consider progressbar 100% complete
+        percent_done = 100;
+    } else if (total_ticks > 0) {
+        percent_done = static_cast<int>(static_cast<float>(ticks) / static_cast<float>(total_ticks) * 100);
```

Comment:
`percent_done` is `int32_t` -> `static_cast<int32_t>`.

Reply:
I used `static_cast<int32_t>`